manifests/base.pp

```diff
@@ -1,8 +1,10 @@
 # base things for shorewall
 class shorewall::base {
 
-  package { 'shorewall':
-    ensure => $shorewall::ensure_version,
+  if ! defined(Package['shorewall']) {
```

Comment:
Why? What if someone defined 'package{'shorewall': ensure => absent }` somewhere else?

Reply:
It’s not easy question =) 

I’m have a problem deploy new node because first time fact shorewall_major_version is empty and define shorewall::rule_section generate wrong config. As a result failed start service shorewall and failed many dependences. 
I solved this by breaking the installation into two steps.
1)	Install packages shorewall
2)	Install class shorewall
Same as:
    if ! defined(Package['shorewall']) {
      package { [shorewall, shorewall6]: }
    }

    if $shorewall_major_version != '' {
      class { 'shorewall': }
    }

But if do it, puppet agent result duplicate declaration Package['shorewall']
It’s crutch, but it’s work.

Reply:
Ah so you rung things twice?

Well you could do the following:

```
if $shorewall_major_version != '' {
  class { 'shorewall': }
} else {
  package { [shorewall, shorewall6]: }
}
```

Reply:
Yes, you are right, this is a more correct solution. Rolled back the change. Thanks!